### PR TITLE
feat: move hosted prod deployment to githubactions

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Move hosted prod deployment to GitHub actions

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by yashasvibajpai — DISMISSED** [review]

**Review by aashishmalik — DISMISSED** [review]

**Review by aashishmalik — DISMISSED** [review]

**Review by yashasvibajpai — DISMISSED** [review]

File: .github/workflows/prepare-for-prod-deploy.yml [hunk 1]
```diff
@@ -197,8 +197,12 @@ jobs:
           yq eval -i ".config-be-user-transformer.image.tag=\"$TAG_NAME\"" values.prod.yaml
           git add values.prod.yaml
 
+          cd ../../customer-objects/multi-tenant-us
+          yq eval -i ".transformer.image.tag=\"$VERSION\"" hostedmtedmt.yaml
+          yq eval -i ".user-transformer.image.tag=\"$VERSION\"" hostedmtedmt.yaml
```
Comment (sanpj2292) [inline .github/workflows/prepare-for-prod-deploy.yml:202]: 1. It seems like we are updating the image of transformer with this command. Is my understanding correct ?
2. Should we update the user-transformer also with the same version as destination-transformer ? Can you once clarify this one ?

Comment (sanpj2292) [reply .github/workflows/prepare-for-prod-deploy.yml:202]: Also it seems like `UT_TAG_NAME` is being formed while creating pull request but I don't see it being used anywhere. Should we remove if it is not being used ?

Comment (Gauravudia) [reply .github/workflows/prepare-for-prod-deploy.yml]: 1. Yes
2. As of now, we are keeping the same tag for both transformer/ user-transformer for all clusters (enterprise, multi-tenant, pro). cc- @aashishmalik 
3. This is for future reference.

Comment (sanpj2292) [reply .github/workflows/prepare-for-prod-deploy.yml]: Got it. Makes sense

**Review by sanpj2292 — APPROVED** [review]

File: .github/workflows/prepare-for-prod-deploy.yml [hunk 2]
```diff
@@ -229,8 +229,12 @@ jobs:
...
 
+          cd ../../customer-objects/multi-tenant-us
+          yq eval -i ".spec.transformer.image.version=\"$TAG_NAME\"" hostedmtedmt.yaml
+          yq eval -i ".spec.user_transformer.image.version=\"$TAG_NAME\"" hostedmtedmt.yaml
+          git add hostedmtedmt.yaml
+
           git commit -m "chore: upgrade shared transformers to $TAG_NAME"
           git push -u origin shared-transformer-$TAG_NAME
 
           hub pull-request -m "chore: upgrade shared transformers to $TAG_NAME"
```
Comment (Thrinadh-Kumpatla) [inline .github/workflows/prepare-for-prod-deploy.yml:240]: Is hub available locally to github action runner instance?

**Review by Thrinadh-Kumpatla — APPROVED** [review]

File: .github/workflows/prepare-for-prod-deploy.yml (hunk 2 above)
Comment (Gauravudia) [reply .github/workflows/prepare-for-prod-deploy.yml:240]: yes

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**